service-base/src/main/java/org/eclipse/hono/service/tenant/CompleteBaseTenantService.java

```diff
@@ -300,15 +300,9 @@ private void addNotPresentFieldsWithDefaultValuesForAdapter(final JsonObject ada
         }
     }
 
-    /**
-     * {@inheritDoc}
-     *
-     * This default implementation simply returns an empty result with status code 501 (Not Implemented).
-     * Subclasses should override this method in order to provide a reasonable implementation.
-     */
     @Override
-    public void add(final Optional<String> tenantId, final JsonObject tenantObj, final Handler<AsyncResult<TenantResult<JsonObject>>> resultHandler) {
-        handleUnimplementedOperation(resultHandler);
+    public void add(Optional<String> tenantId, JsonObject tenantObj, Handler<AsyncResult<CreateResult>> resultHandler) {
+        resultHandler.handle(Future.failedFuture(new ServerErrorException(HttpURLConnection.HTTP_NOT_IMPLEMENTED)));
```

Comment:
what's wrong with *handleUnimplementedOperation*?

Reply:
@dejanb  defined a _CreateResult_ class to return the resource Version to the client, as defined in the new spec. This modifies the type of the handler to `Handler<AsyncResult<CreateResult>>` so _handleUnimplementedOperation_ is not compatible anymore as it requires `Handler<AsyncResult<TenantResult<JsonObject>>>_